node/src/components/chain_synchronizer/operations.rs

```diff
@@ -866,14 +929,19 @@ pub(super) async fn run_chain_sync_task(
             Some(block_with_metadata) => block_with_metadata.block,
         };
 
-        let mut deploys: Vec<Deploy> = Vec::with_capacity(block.deploy_hashes().len());
-        for deploy_hash in block.deploy_hashes() {
-            deploys.push(*fetch_and_store_deploy(effect_builder, *deploy_hash).await?);
-        }
-        let mut transfers: Vec<Deploy> = Vec::with_capacity(block.transfer_hashes().len());
-        for transfer_hash in block.transfer_hashes() {
-            transfers.push(*fetch_and_store_deploy(effect_builder, *transfer_hash).await?);
-        }
+        let deploys = fetch_and_store_deploys(
+            block.deploy_hashes().to_vec(),
+            max_parallel_deploy_fetches,
+            effect_builder,
+        )
+        .await?;
+
+        let transfers = fetch_and_store_deploys(
+            block.transfer_hashes().to_vec(),
+            max_parallel_deploy_fetches,
+            effect_builder,
+        )
+        .await?;
```

Comment:
```let (deploys, transfers) = tokio::join!(..., ...)```
would probably improve things even further.

Reply:
Or ```futures::future::join```

Reply:
Or just pass the combined list into a single call of `fetch_and_store_deploys` instead of making two separate calls. That's what we do in `sync_deploys_and_transfers_and_state`.

Reply:
Unlike in `sync_deploys_and_transfers_and_state`, here we need a separate lists of `deploys` and `transfers` as an output (they are used later). We could probably read them all at once and the partition (assuming a deploy can identify itself as transfer or not).

As mentioned in the PR description, there are better ways. The time was short, though, so was the scope of the PR different :)